addons/web/views/report_templates.xml

```diff
@@ -406,12 +406,19 @@
         </div>
 
         <div t-attf-class="footer o_company_#{company.id}_layout {{report_type != 'pdf' and 'mt-auto'}}">
-            <div class="o_footer_content row border-top pt-2">
+            <div t-if="is_html_empty(company.report_header)" class="o_footer_content d-flex border-top pt-2">
+                <div class="flex-grow-1 text-start me-2" t-field="company.report_footer"/>
+                <div class="text-end text-muted">
+                    <div t-if="report_type == 'pdf' and display_name_in_footer" t-out="o.name">(document name)</div>
+                    <div t-if="report_type == 'pdf'">Page <span class="page"/> / <span class="topage"/></div>
+                </div>
+            </div>
+            <div t-else="" class="o_footer_content row border-top pt-2">
                 <div class="col-8">
                     <span t-field="company.report_footer"/>
                 </div>
                 <div class="col-4 text-end">
-                    <strong t-if="company.report_header" t-field="company.report_header" class="o_company_tagline">Company tagline</strong>
+                    <strong t-field="company.report_header" class="o_company_tagline">Company tagline</strong>
```

Comment:
Thanks a lot for fixing this. I suggest we do this instead to avoid duplication. 
```xml
<div t-attf-class="{{'flex-grow-1' if is_html_empty(company.report_header) else 'col-8'}}">
     <span t-field="company.report_footer"/>
</div>
<div t-attf-class="{{'' if is_html_empty(company.report_header) else 'col-4'}} text-end">
    ...
</div>
```

I briefly tested and it worked fine, but I'm on an ARM processor (mac) and wkhtmltopdf is not supported well on these architecture so my rendering is not reliable.

Reply:
You got an overlap of page number and footer in those case.
![image](https://github.com/user-attachments/assets/31baf57a-1567-4b10-a2af-02a55b5d7015)
With the current version you got another result
![image](https://github.com/user-attachments/assets/bc15abe0-4161-467c-a53f-d215fe328c19)

But it's probably fixable with a css class but I'm really bad at knowing them :)
@mano-odoo

Reply:
Done. I slightly modify it but kept the idea.
pdf (without tag line)
![image](https://github.com/user-attachments/assets/6b6c030f-63ce-4cfd-9a75-83fabf561929)

pdf (with tag line) -> The smile emoji)
![image](https://github.com/user-attachments/assets/945bbf5f-0606-4786-9396-0fcc5aee02b0)


Before
![image](https://github.com/user-attachments/assets/0f832ebe-2b05-4c78-82c2-9047f8385714)